facebook/src/com/facebook/GraphRequest.java

```diff
@@ -603,22 +603,24 @@ public static GraphRequest newCustomAudienceThirdPartyIdRequest(AccessToken acce
                 AttributionIdentifiers.getAttributionIdentifiers(context);
         Bundle parameters = new Bundle();
 
-        if (accessToken == null) {
-            // Only use the attributionID if we don't have an access token.  If we do, then the user
-            // token will be used to identify the user, and is more reliable than the attributionID.
-            String udid = attributionIdentifiers.getAttributionId() != null
-                    ? attributionIdentifiers.getAttributionId()
-                    : attributionIdentifiers.getAndroidAdvertiserId();
-            if (attributionIdentifiers.getAttributionId() != null) {
-                parameters.putString("udid", udid);
+        if (attributionIdentifiers != null) { //check if attributionIdentifiers is null to prevent NPEs
```

Comment:
This isn't quite right either.

You want to check them when the accessToken is null, and if they're also null then throw as we have no data to create this request:

     if (accessToken == null) {
            if (attributionIdentifiers == null) {
                throw new FacebookException(
                        "There is no access token and attribution identifiers could not be " +
                                "retrieved");
            }

And then below, protect against them being null:

	​​        if (FacebookSdk.getLimitEventAndDataUsage(context)
                || (attributionIdentifiers != null && attributionIdentifiers.isTrackingLimited())) {
            parameters.putString("limit_event_usage", "1");
        }